internal/controller/services/gateway/ocp_routes.go

```diff
@@ -0,0 +1,147 @@
+// Format: <gateway-name>-<gatewayclass-name>.
+var GatewayServiceFullName = DefaultGatewayName + "-" + GatewayClassName
+
+// httpRouteReferencesGateway checks if an HTTPRoute references our gateway.
+func httpRouteReferencesGateway(httpRoute *gwapiv1.HTTPRoute) bool {
+	for _, ref := range httpRoute.Spec.ParentRefs {
+		refNamespace := GatewayNamespace
+		if ref.Namespace != nil {
+			refNamespace = string(*ref.Namespace)
+		}
+		if string(ref.Name) == DefaultGatewayName && refNamespace == GatewayNamespace {
+			return true
+		}
+	}
+	return false
+}
+
+// HTTPRouteGatewayRefPredicate filters HTTPRoutes that reference our gateway.
+func HTTPRouteGatewayRefPredicate() predicate.Funcs {
+	return predicate.Funcs{
+		CreateFunc: func(e event.CreateEvent) bool {
+			httpRoute, ok := e.Object.(*gwapiv1.HTTPRoute)
+			if !ok {
+				return false
+			}
+			return httpRouteReferencesGateway(httpRoute)
+		},
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			httpRoute, ok := e.ObjectNew.(*gwapiv1.HTTPRoute)
+			if !ok {
+				return false
+			}
+			return httpRouteReferencesGateway(httpRoute)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			httpRoute, ok := e.Object.(*gwapiv1.HTTPRoute)
+			if !ok {
+				return false
+			}
+			return httpRouteReferencesGateway(httpRoute)
+		},
+		GenericFunc: func(e event.GenericEvent) bool {
+			httpRoute, ok := e.Object.(*gwapiv1.HTTPRoute)
+			if !ok {
+				return false
+			}
+			return httpRouteReferencesGateway(httpRoute)
+		},
+	}
+}
+
+// createOCPRoutes creates a single OCP Route for the Gateway when in OcpRoute mode.
+func createOCPRoutes(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	l := logf.FromContext(ctx).WithName("createOCPRoutes")
+
+	gatewayConfig, err := validateGatewayConfig(rr)
+	if err != nil {
+		return err
+	}
+
+	if gatewayConfig.Spec.IngressMode != serviceApi.IngressModeOcpRoute {
+		l.V(1).Info("IngressMode is not OcpRoute, skipping OCP Route creation")
+		return nil
+	}
+
+	fqdn, err := GetFQDN(ctx, rr.Client, gatewayConfig)
+	if err != nil {
+		return fmt.Errorf("failed to get FQDN for OCP Route: %w", err)
+	}
+
+	weight := int32(100)
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      DefaultGatewayName,
+			Namespace: GatewayNamespace,
+			Labels: map[string]string{
+				labels.PlatformPartOf: PartOfGatewayConfig,
+			},
+			Annotations: map[string]string{
+				"router.openshift.io/service-ca-certificate": "true",
+			},
+		},
+		Spec: routev1.RouteSpec{
+			Host: fqdn,
+			To: routev1.RouteTargetReference{
+				Kind:   "Service",
+				Name:   GatewayServiceFullName,
+				Weight: &weight,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromInt(StandardHTTPSPort),
+			},
+			TLS: &routev1.TLSConfig{
+				Termination:                   routev1.TLSTerminationReencrypt,
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
+			},
+		},
+	}
```

Comment:
shall we create a template for Route, just like HttpRoute, internal/controller/services/gateway/resources/kube-auth-proxy-httproute.tmpl.yaml